packages/orchestrator-ui-components/src/components/WfoTable/WfoTableWithFilter/updateQueryString.ts

```diff
@@ -0,0 +1,52 @@
+const getValueWithoutBrackets = (value: string) => {
+    if (value.startsWith('(') && value.endsWith(')')) {
+        return value.slice(1, -1);
+    }
+    return value;
+};
+
+const isContainingSpaces = (value: string) => {
+    return value.indexOf(' ') !== -1;
+};
+
+const toQueryValue = (value: string) => {
+    return isContainingSpaces(value) ? `"${value}"` : value;
+};
+
+// Three cases to consider when this function is called
+// 1 - queryString is empty
+// 2 - fieldName already exists in query string
+// 3 - fieldName does not exist in query string
+export const updateQueryString = (
+    queryString: string | undefined,
+    fieldName: string,
+    value: string,
+) => {
+    // 1 - Empty query string returning directly fieldName:value
+    if (!queryString) {
+        return `${fieldName}:${toQueryValue(value)}`;
+    }
+
+    const fieldRegex = new RegExp(
+        `${fieldName}:(\\([\\S "]+\\)|"[\\S ]+"|\\S+)`,
```

Comment:
- maybe better to use \s instead of space
- You can probably combine the last two as just | [\\S\s]+

Reply:
Nice one, but are there unforseen characters that we want to catch with `\s`?

Reply:
In python I usually use "raw" strings to avoid having to escape every backslash, for example instead of `"\\d+\\s"` I'd write `r"\d+\s"` (where the 'r' prefix means raw).

It looks like javascript's equivalent for this is `String.raw`. Maybe you can use this to make the part after the ":" slightly more readable?